src/System.Linq.Expressions/tests/ExceptionHandling/ExceptionHandlingExpressions.cs

```diff
@@ -137,6 +137,108 @@ public void CanCatchAndThrowNonExceptionsInterpreted()
         }
 
         [Fact]
+        public void ThrowNullThrowsNRE()
+        {
+            Action throwNull = Expression.Lambda<Action>(
+                Expression.Throw(Expression.Constant(null, typeof(Expression)))
+                ).Compile(false);
+            Assert.Throws<NullReferenceException>(throwNull);
+        }
+
+        [Fact]
+        public void ThrowNullThrowsNREInterpreted()
+        {
+            Action throwNull = Expression.Lambda<Action>(
+                Expression.Throw(Expression.Constant(null, typeof(Expression)))
+                ).Compile(true);
+            Assert.Throws<NullReferenceException>(throwNull);
+        }
```

Comment:
Okay. Doing that with `[InlineData]` seem really obvious now that you've pointed it out!